src/filter-clients.ts

```diff
@@ -0,0 +1,37 @@
+import * as clients from '../clients'
+import {Client} from '../interfaces/client.interface'
+import {RegistryEntry} from '../interfaces/registry.interface'
+import {getValueForEnvironment} from './utils'
+
+const transformClientObject = (client: Client, environment: string): RegistryEntry => {
+    const clientId = getValueForEnvironment(environment, client.clientId)
+    return {
+        clientId,
+        clientType: client.clientType,
+        isAllowed: client.isAllowed,
+        isHmrc: client.isHmrc,
+        isReportSuspiciousActivityEnabled: client.isReportSuspiciousActivityEnabled,
+        isAvailableInWelsh: client.isAvailableInWelsh,
+        showInClientSearch: getValueForEnvironment(environment, client.showInClientSearch),
+    }
+}
+
+const filterClients = (environment: string, filter: Record<string, any>): RegistryEntry[] => {
+    const filteredClients: Client[] = Object.values(clients).filter(client => {
+        return Object.keys(filter).some(key => {
+            return client[key as keyof Client] === filter[key];
+        });
+    });
```

Comment:
could this be made more functional?

```suggestion
const filterClients = (environment: string, filter: Record<string, any>): RegistryEntry[] => {
    return Object.values(clients).filter(client => {
        return Object.keys(filter).some(key => {
            return client[key as keyof Client] === filter[key];
        });
    }).map((client) => transformClientObject(client, environment)
```

Reply:
Yup, will update as well.